src/discordcr/snowflake.cr

```diff
@@ -0,0 +1,48 @@
+module Discord
+  DISCORD_EPOCH = 1420070400000_u64
+
+  # Struct representing a Discord ID
+  struct Snowflake
+    getter value : UInt64
+
+    # The time at which this snowflake was created
+    getter creation_time : Time do
+      ms = (value >> 22) + DISCORD_EPOCH
+      Time.epoch_ms(ms)
+    end
+
+    def self.new(string : String)
+      new(string.to_u64)
+    end
+
+    def self.new(parser : JSON::PullParser)
+      string = parser.read_string
+      new(string.to_u64)
+    end
+
+    # Creates a `Snowflake` embedded with the given timestamp
+    def self.new(time : Time)
```

Comment:
I don't like this constructor because there's no canonical snowflake for a given timestamp.

What's the usecase for this ctor instead of just using `creation_time`.

Reply:
`REST#get_channel_messages` before/after/around a specific date (and eventuallyTM the search endpoint)

Reply:
Additionally, in bulk deleting messages, it can be used to calculate the minimum snowflake so that you don't request messages to be deleted that Discord will reject. (You can't bulk-delete messages older than two weeks old)

Reply:
k